### PR TITLE
[INT-292] New Canva Button integration

### PR DESCRIPTION
**Pull request by cmunozgar**
### Description
This PR adds new integration with [Canva](https://www.canva.com) through a new `DropdownMenu` in the `UploaderZone` component.

Useful documentation:
- [Canva Local Development](https://paper.dropbox.com/doc/Canvas-Local-Development--BPRjHbf60BcBTEjsf17IKI0AAg-ji2mi0XRmT6emapYa8pIs)
- [Design Brief](https://paper.dropbox.com/doc/Design-Brief-Canva-Integration--BNk5fqqbX7T~WBpHvEiVA7RTAg-WXCDI4iNzgqcnVinWhQmH)
- [Canva API Documentation](https://docs.developer.canva.com/button/javascript/getting-started)

### Context & Notes
If you go through the massive commits during the development of this feature you'll observe that initially, we created a new `CanvaButton` component to place it within the composers. Finally, we decided to add a new dropdown menu on the uploader to give the customer the ability to select to upload files from their computer or with new integrations (Canva is the initial one).

That's why there are no .jsx files within the new `canva` packages folder. The file `handlers.js` contains all the business logic to open (and edit #1851) a new Canva modal. The `constants.js` file is self-explanatory.

**Notes for Engineers**
Even though the PR has **19 file changes**, most of the file changes are simple like:
- Configuration for loading the Canva SDK javascript files
  - `packages/server/index.ejs`
  - `webpack.config.branch.js`
  - `webpack.config.dev.js`
  - `webpack.config.prod.js`
  - `webpack.config.staging.js`
- Updating the @bufferapp/ui components library
  - `package.json`
  - `yarn.lock`
- Drilldown of the feature-flip from the User to the UploadZone component
  - `packages/composer/composer/components/AppStateless.jsx`
  - `packages/composer/composer/components/Composer.jsx`
  - `packages/composer/composer/components/ComposerSection.jsx`
  - `packages/composer/composer/stores/AppStore.js`
  - `packages/composer/composer/utils/DataImportUtils.js`
  - `packages/server/parsers/userParser.js`

The approach that I took was to place the new dropdown menu inside the `UploadZone` component. And, specify a new prop `showIntegrationsDropdown` as the feature flag. You will see that the integrations dropdown is enabled if you have the `canva_button` feature flip.

Inside the `UploadZone` you will find a new component named `UploadMessage` that will help us change to add the new prompt message for the uploader. We are now making it clearer to the customer that you can drag & drop files to that container or click and select the file. This change will be displayed to all customers, regardless of the `canva_button` feature flip.
 
**How to call Canva?**
We need to specify the following parameters to the function `openCanvaModal`:

- draftId
- serviceId
- notifiers 
- uploadDraftFile
- designId (for the edit functionality)
  
So the main changes are in the remaining files.

**Composer colors**
You might also note that the Publish Composer (old composer) has changed its colors from `#1DA1F2` to `#2C4BFF`.

### Screenshots (if appropriate)
Screenshot 2021-08-02 at 11 38 32

Screenshot 2021-08-02 at 11 38 41

Screenshot 2021-08-02 at 11 38 47

Screenshot 2021-08-02 at 11 38 56

Screenshot 2021-08-02 at 11 39 04

Screenshot 2021-08-02 at 11 39 23

Screenshot 2021-08-02 at 11 39 30

Screenshot 2021-08-02 at 11 39 16

Screenshot 2021-08-02 at 11 38 26

### Best Practices Checklist

-   [X] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] I have tested different user stories.
-   [X] I have identified similar or related features and tested they are still working.
-   [X] I have performed a self-review of my own code.
-   [X] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [X] I have commented on my code, particularly in hard-to-understand areas.
-   [X] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

### ⚠️ Pre-deploy Checklist

- [X] I have manually loaded the URL of the staging deployment and confirmed it's working (see below**)
- [X] I have confirmed that the Cypress E2E test is passing (_Note: we have a known issue where it doesn't run on some PRs. Hopefully this will be fixed soon._)
- [X] I have confirmed that all other tests are passing.

** To visit the URL of the staging deployment, please click the link in the bot comment, or look under checks at the bottom of this PR and click "Details" next to `bufferbotbrains/cicd-buffer-publish`
image

### CODE REVIEW

File: packages/UploadZone/components/UploadZone/index.jsx [hunk 1]
```diff
@@ -44,7 +44,6 @@ const ButtonWithStyles = styled(Button)`
 
 const DropzoneWithStyles = styled(Dropzone)`
   cursor: pointer;
-  position: absolute;
```
Comment (cmunozgar) [inline packages/UploadZone/components/UploadZone/index.jsx:47]: To be able to click the Canva button inside the UploadZone. I wonder if there's a better way to do this.

**Review by MayaUribe — COMMENTED** [review]
So excited for the Canva integration @cmunozgar  🤩

I added a few small comments 😊

**Some comments:**
- We should confirm with Lauren that the composer upload icons would change for all users.
- Perhaps it'd be worth having a test plan to ensure all the places where we upload are working as expected with and without the FF (upload on each channel, IG stories, drafts, etc) just to confirm we went through all the different scenarios.

File: packages/story-group-composer/components/Carousel/CardItem/index.jsx [hunk 2]
```diff
@@ -131,10 +125,14 @@ class CardItem extends React.Component {
...
                 maxAttachableImagesCount: maxAttachableMediaCount,
                 canHaveMediaAttachmentType: () => true,
+                name: 'instagramStory', // Used by Canva Integration
               }}
               uploadType={UploadTypes.MEDIA}
               multiple
               disabled={isOver}
+              showIntegrationsDropdown={shouldShowCanvaButton}
+              showUploadMessage={!card.uploading}
+              showUploadText={true}
```
[the remark this answers is not in the document]
Comment (cmunozgar) [reply packages/story-group-composer/components/Carousel/CardItem/index.jsx:135]: I can flip upside down the default value of `showUploadText` for UploadZone to make it simpler.

File: packages/UploadZone/components/UploadZone/index.jsx [hunk 3]
```diff
@@ -236,11 +308,21 @@ class UploadZone extends React.Component {
...
+            ariaLabelPopup="Help"
+            menubarItem={uploadingButton}
+            items={menuItems}
+            xPosition="left"
+            ref={node => {
+              this.dropdownmenu = node;
+            }}
+          />
+        )}
+        {!showIntegrationsDropdown && uploadingButton}
```
[the remark this answers is not in the document]
Comment (cmunozgar) [reply packages/UploadZone/components/UploadZone/index.jsx]: Suggestion applied!

**Review by anafilipadealmeida — COMMENTED** [review]
Thank you so much @cmunozgar, there's a lot of work here! Left some small comments 🙂

File: packages/composer/composer/components/css/constants.css [hunk 4]
```diff
@@ -8,9 +8,9 @@
   tertiary-font-color: color(#59626a lightness(60%));
   app-border-color: #d5dce3;
   app-secondary-border-color: #f8f8f8;
-  blue-color-primary-hue: #168eea;
+  blue-color-primary-hue: #2C4BFF;
   blue-color-darker-hue: #0f63a4;
-  blue-color-secondary-hue: #137ac9;
+  blue-color-secondary-hue: #121E66;
```
[the remark this answers is not in the document]
Comment (cmunozgar) [reply packages/composer/composer/components/css/constants.css:13]: Good catch!

File: packages/server/parsers/userParser.js [hunk 5]
```diff
@@ -32,13 +32,15 @@ module.exports = userData => ({
...
+      'billingNotifications'
+    ),
   },
   analyzeCrossSale: userData.is_analyze_customer,
   features: userData.features, // these are user features only, for org features (owner features) check org parser
   hasCalendarFeature: userData.features.includes('new_calendar'), // for dev purposes only, to be replaced by an entitlement on the release date
   hasCreatePostButtonFeature: userData.features.includes(
     'create_post_button_queue'
   ), // this is for experimenting with the button in the queue
+  canva_button: userData.features.includes('canva_button'),
```
[the remark this answers is not in the document]
Comment (cmunozgar) [reply packages/server/parsers/userParser.js:45]: let me change it 👍🏼

File: packages/UploadZone/components/UploadMessage/index.jsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    `}
+  span {
+    color: ${Colors.blue};
+    text-decoration: none;
+  }
+  &:hover {
+    svg {
+      fill: ${Colors.blue};
+    }
+    a {
```
[the remark this answers is not in the document]
Comment (cmunozgar) [reply packages/UploadZone/components/UploadMessage/index.jsx:34]: No, this should be removed! Good catch!

File: packages/UploadZone/components/UploadMessage/index.jsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+const Message = styled.div`
+  position: absolute;
+  top: 50%;
+  left: 0;
+  right: 0;
+  transform: translate(0, -50%);
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  font-size: 14px;
```
[the remark this answers is not in the document]
Comment (cmunozgar) [reply packages/UploadZone/components/UploadMessage/index.jsx]: I will make sure to add this component in the next PR. In the meantime, I'm removing the unnecessary `font-size` and `line-height`

File: packages/server/parsers/userParser.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (cmunozgar) [reply packages/server/parsers/userParser.js]: I'm leaving the snake_case only for the feature. Everything else has been changed to camelCase

**Review by anafilipadealmeida — APPROVED** [review]

**Pull request merged**